src/main/java/cz/muni/fi/pv286/mmt/arguments/ArgParser.java

```diff
@@ -0,0 +1,504 @@
+    }
+
+    private void setOption(Options options, FromToOption fromToOption, FromOrTo fromOrTo)
+            throws BadArgumentsException {
+        try {
+            if (FromOrTo.FROM == fromOrTo) {
+                options.setInputFromToOption(fromToOption);
+            } else if (FromOrTo.TO == fromOrTo) {
+                options.setOutputFromToOption(fromToOption);
+            } else {
+                throw new BadArgumentsException(UNKNOWN_OPTION);
+            }
+        } catch (IllegalArgumentException e) {
+            throw new BadArgumentsException(UNKNOWN_OPTION);
+        }
+
+    }
+
+    private void setFormat(Options options, IoFormat format, FromOrTo fromOrTo)
+            throws BadArgumentsException {
+        if (this.hasFromFormat && (fromOrTo == FromOrTo.FROM)
+                || this.hasToFormat && (fromOrTo == FromOrTo.TO)) {
+            throw new BadArgumentsException(FORMAT_SET_DUPLICITY);
+        }
+        if (FromOrTo.FROM == fromOrTo) {
+            this.hasFromFormat = true;
+            options.setInputFormat(format);
+        } else if (FromOrTo.TO == fromOrTo) {
+            this.hasToFormat = true;
+            options.setOutputFormat(format);
+        }
+    }
+
+    private void setFile(Options options, String path, FromOrTo fromOrTo)
+            throws BadArgumentsException, FileNotFoundException, InvalidInputException {
+        if (this.hasInputFile && (fromOrTo == FromOrTo.FROM)
+                || this.hasOutputFile && (fromOrTo == FromOrTo.TO)) {
+            throw new BadArgumentsException(FILE_SET_DUPLICITY);
+        }
+        File file = new File(path);
+        if (file.isDirectory()) {
+            throw new BadArgumentsException(FILE_IS_DIRECTORY);
+        }
+        if (!file.canWrite() && fromOrTo == FromOrTo.TO) {
+            throw new InvalidInputException(CANT_WRITE_FILE);
+        }
+        if (!file.canRead() && fromOrTo == FromOrTo.FROM) {
+            throw new InvalidInputException(CANT_READ_FILE);
+        }
+        if (FromOrTo.FROM == fromOrTo) {
+            this.hasInputFile = true;
+            options.setInputFile(new FileInputStream(file));
+        } else if (FromOrTo.TO == fromOrTo) {
+            this.hasOutputFile = true;
+            options.setOutputFile(new FileOutputStream(file));
+        }
+    }
+
+    private void setDelimiter(Options options, String delimiter)
+            throws BadArgumentsException {
+        if (this.hasDelimiter) {
+            throw new BadArgumentsException(MESSAGE_INVALID_ARGUMENTS);
+        }
+        this.hasDelimiter = true;
+        options.setDelimiter(delimiter);
+    }
+
+    private void checkForHelp(String[] args)
+            throws HelpInvokedException {
+        for (String arg : args) {
+            if (Patterns.shortHelp.matcher(arg).matches()
+                    || Patterns.longHelp.matcher(arg).matches()) {
+                throw new HelpInvokedException();
```

Comment:
**issue:** throwing exception when no error has occured

Throwing an exception when there is no mistake in program arguments doesn't make sense.
**suggestion:** Consider creating a bool variable to remember if program should print help.